appcues/src/main/java/com/appcues/ui/composables/StepComposition.kt

```diff
@@ -119,11 +125,31 @@ private fun Step.ComposeStickyContent(
     }
 }
 
-// conditionally add vertical scrolling to the step, only if the given ScrollState is not null.
-// this is used to allow the parent trait to control whether or not step content is scrollable
-private fun Modifier.stepVerticalScroll(enabled: Boolean) = composed {
+// The `enabled` parameter allows the parent trait to control whether or not step content is
+// scrollable. Embeds, for instance, would not allow vertical scrolling, as they just size the
+// content to the required height.
+//
+// If enabled, the maxHeight param is used to check if the sized content will actually need
+// scrolling. If the content height is less than the maxHeight, then no scrolling is enabled. This
+// is so that other gestures like swipe dismiss can be applied.
+//
+// If enabled and the content is taller than the available space, then scrolling is enabled.
+private fun Modifier.stepVerticalScroll(enabled: Boolean, maxHeightDp: Dp) = composed {
     if (enabled) {
-        Modifier.verticalScroll(rememberScrollState())
+        val contentHeight = remember { mutableIntStateOf(0) }
+        val maxHeight = with(LocalDensity.current) { maxHeightDp.toPx() }
+        Modifier
+            // Measures the height of the content to determine if scroll needed
+            .onGloballyPositioned { coordinates ->
+                contentHeight.intValue = coordinates.size.height
+            }
+            .let {
+                if (contentHeight.intValue < maxHeight) {
+                    it
+                } else {
+                    it.verticalScroll(rememberScrollState())
+                }
+            }
```

Comment:
This is the change for the vertical scroll handling - conditionally adding scroll only if the container height is not less than max available height. It is more complex to measure the actual content within, but this container measurement should provide effectively the same result.

Reply:
snapshot tests are showing there is something not quite right here as well with content height sizing, looking into.